maintenance/ReloadEnwikiDump.php

```diff
@@ -0,0 +1,219 @@
+	$IP = __DIR__ . '/../../..';
+}
+
+require_once "$IP/maintenance/Maintenance.php";
+
+class ReloadEnwikiDump extends Maintenance {
+
+	private const UPDATELOG_KEY = 'WikiMirror:ReloadEnwikiDump:Version';
+
+	public function __construct() {
+		parent::__construct();
+		$this->requireExtension( 'WikiMirror' );
+		$this->setBatchSize( 50000 );
+		$this->addDescription(
+			'Update locally-stored tracking data about which NS_MAIN pages exist on the English Wikipedia.'
+		);
+
+		$this->addOption(
+			'dump',
+			'Date for the dump to import from the English Wikipedia',
+			true,
+			true
+		);
+	}
+
+	/**
+	 * Update remote_page table contents if needed
+	 * @return bool
+	 */
+	public function execute() {
+		$dump = $this->getOption( 'dump' );
+		$current = $this->getCurrentDumpVersion();
+		if ( $dump === $current ) {
+			$this->output( "Dump is already using version $dump" );
+			return true;
+		}
+		$dumpPath = $this->downloadDump( $dump );
+		// $dumpPath = "/tmp/WikiMirror-ReloadEnwikiDump-bHKsBr";
+		$sqlPath = $this->runFirstLoad( $dumpPath );
+		// $sqlPath = "/tmp/WikiMirror-ReloadEnwikiDump-sql-B0KyJV";
+		$this->runSQLLoad( $sqlPath );
+		$this->runFinish();
+
+		unlink( $dumpPath );
+		unlink( $sqlPath );
+
+		$this->storeDumpVersion( $dump );
+		return true;
+	}
+
+	/**
+	 * Get the current version of the dump that is loaded based on the updatelog
+	 * table, or null if not stored
+	 *
+	 * @return string|null
+	 */
+	private function getCurrentDumpVersion(): string|null {
+		$query = $this->getPrimaryDB()
+			->newSelectQueryBuilder()
+			->select( 'ul_value' )
+			->from( 'updatelog' )
+			->where( [ 'ul_key' => self::UPDATELOG_KEY ] )
+			->caller( __METHOD__ );
+		$result = $query->fetchField();
+		if ( $result === false ) {
+			return null;
+		}
+		return $result;
+	}
+
+	/**
+	 * Update the stored version of the dump that is loaded in updatelog
+	 *
+	 * @param string $version
+	 */
+	private function storeDumpVersion( string $version ) {
+		$query = $this->getPrimaryDB()
+			->newInsertQueryBuilder()
+			->insertInto( 'updatelog' )
+			->row( [
+				'ul_key' => self::UPDATELOG_KEY,
+				'ul_value' => $version
+			] )
+			->onDuplicateKeyUpdate()
+			->uniqueIndexFields( [ 'ul_key' ] )
+			->set( [ 'ul_value' => $version ] )
+			->caller( __METHOD__ );
+		$query->execute();
+		$this->output( "updatelog table updated, dump is $version\n" );
+	}
+
+	/**
+	 * Download the dump from
+	 * `https://dumps.wikimedia.org/enwiki/{id}/enwiki-{id}-all-titles-in-ns0.gz`
+	 * and put it in /tmp
+	 */
+	private function downloadDump( string $id ): string {
+		$services = $this->getServiceContainer();
+		$httpFactory = $services->getHTTPRequestFactory();
+		$reqUrl = "https://dumps.wikimedia.org/enwiki/$id/enwiki-$id-all-titles-in-ns0.gz";
+		$request = $httpFactory->create(
+			$reqUrl,
+			[
+				'method' => 'GET',
+				// 5 minutes should be enough? Got a timeout with 3
+				'timeout' => 60 * 5,
+				'maxTimeout' => 60 * 5,
```

Comment:
Suggesting set high enough `connectTimeout` to ensure network conditions won't halt the process